blockchain/processing.py

```diff
@@ -235,17 +275,21 @@ def _execute_phase_1(self, config, current_block_id):
         print ("""Time bounds: %i - %i""" % (block_bound_lower_ts, block_bound_lower_ts + BLOCK_INTERVAL))
         transaction_db.fixate_block(block_bound_lower_ts, block_bound_lower_ts + BLOCK_INTERVAL, current_block_id)
 
-        if 'approve_block' in config:
-            return config['approve_block'](config, current_block_id)
-
         transactions = transaction_db.get_all(block_id=current_block_id)
 
         # Validate the schema and structure of the transactions
         valid_transactions, invalid_transactions = self.split_items(valid_transaction_sig, transactions)
 
+        # TODO: Take this condition out?
+        if 'approve_block' in config:
+            # TODO: strip reserved transaction types. pass stripped list instead of config
+            # TODO: call reserved transaction function
+            return config['approve_block'](config, current_block_id)
+
         # Use the custom approval code if configured, otherwise approve all valid transaction
         rejected_transactions = []
         if 'approve_transaction' in config:
```

Comment:
Refactor this conditional as well - 

There should no longer be a `approve_transaction` configuration item, instead, we should be calling whatever smart contract is registered for that transaction type.

If none are registered, it should be denied (kill lines 295-296).

Let me know if you have questions.